src/Xamarin.Android.Build.Tasks/Tasks/ResolveAndroidTooling.cs

```diff
@@ -0,0 +1,396 @@
+﻿using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.Linq;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Globalization;
+
+namespace Xamarin.Android.Tasks
+{
+	/// <summary>
+	/// ResolveAndroidTooling does lot of the grunt work ResolveSdks used to do:
+	/// - Modify TargetFrameworkVersion
+	/// - Calculate ApiLevel and ApiLevelName
+	/// - Find the paths of various Android tooling that other tasks need to call
+	/// </summary>
+	public class ResolveAndroidTooling : Task
+	{
+		public string AndroidNdkPath { get; set; }
+
+		public string AndroidSdkPath { get; set; }
+
+		public string AndroidSdkBuildToolsVersion { get; set; }
+
+		public string ProjectFilePath { get; set; }
+
+		public string SequencePointsMode { get; set; }
+
+		public bool UseLatestAndroidPlatformSdk { get; set; }
+
+		public bool AotAssemblies { get; set; }
+
+		[Output]
+		public string TargetFrameworkVersion { get; set; }
+
+		[Output]
+		public string AndroidApiLevel { get; set; }
+
+		[Output]
+		public string AndroidApiLevelName { get; set; }
+
+		[Output]
+		public string SupportedApiLevel { get; set; }
+
+		[Output]
+		public string AndroidSdkBuildToolsPath { get; set; }
+
+		[Output]
+		public string AndroidSdkBuildToolsBinPath { get; set; }
+
+		[Output]
+		public string ZipAlignPath { get; set; }
+
+		[Output]
+		public string AndroidSequencePointsMode { get; set; }
+
+		[Output]
+		public string LintToolPath { get; set; }
+
+		[Output]
+		public string ApkSignerJar { get; set; }
+
+		[Output]
+		public bool AndroidUseApkSigner { get; set; }
+
+		[Output]
+		public bool AndroidUseAapt2 { get; set; }
+
+		[Output]
+		public string Aapt2Version { get; set; }
+
+		static readonly bool IsWindows = Path.DirectorySeparatorChar == '\\';
+		static readonly string ZipAlign = IsWindows ? "zipalign.exe" : "zipalign";
+		static readonly string Aapt = IsWindows ? "aapt.exe" : "aapt";
+		static readonly string Aapt2 = IsWindows ? "aapt2.exe" : "aapt2";
+		static readonly string Android = IsWindows ? "android.bat" : "android";
+		static readonly string Lint = IsWindows ? "lint.bat" : "lint";
+		static readonly string ApkSigner = "apksigner.jar";
+
+		public override bool Execute ()
+		{
+			Log.LogDebugMessage ("ResolveAndroidTooling:");
+			Log.LogDebugMessage ("  TargetFrameworkVersion: {0}", TargetFrameworkVersion);
+			Log.LogDebugMessage ("  AndroidApiLevel: {0}", AndroidApiLevel);
+			Log.LogDebugMessage ($"  {nameof (AndroidSdkPath)}: {AndroidSdkPath}");
```

Comment:
if we are refactoring , should we use `nameof` for all the parameters?

Reply:
Yes, please!

Reply:
Although...  Do we *actually* need all of this anymore?

MSBuild will print out all of this information now:

```
  Task "Delete" (TaskId:391)
    Task Parameter:Files=…/Xamarin.Forms_Performance_Integration-Signed-Unaligned.apk (TaskId:391)
    File "…/Xamarin.Forms_Performance_Integration-Signed-Unaligned.apk" doesn't exist. Skipping. (TaskId:391)
```

The problem, IIRC, was that *`xbuild`* didn't log out this information, which necessitated that *we* provide the information if we wanted it.

Now that we're using `msbuild` everywhere, I'm not sure that we need to cater to `xbuild` anymore.  As such, do we actually need *any* of this messaging?

Reply:
Yeah here is an example of a diagnostic log:
```
Using "InstallPackageAssemblies" task from assembly "/Users/builder/data/lanes/6090/26a17a14/source/monodroid/external/xamarin-android/bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/Xamarin.Android.Build.Debugging.Tasks.dll".
Task "InstallPackageAssemblies" (TaskId:214)
  Task Parameter:AaptToolPath=/Users/builder/android-toolchain/sdk/build-tools/27.0.3/ (TaskId:214)
  Task Parameter:ApkFileSigned=bin/Debug/UnnamedProject.UnnamedProject-Signed.apk (TaskId:214)
  Task Parameter:EmbedAssemblies=False (TaskId:214)
  Task Parameter:PreserveUserData=True (TaskId:214)
  Task Parameter:PackageName=UnnamedProject.UnnamedProject (TaskId:214)
  Task Parameter:
      ResolvedAssemblies=
          obj/Debug/android/assets/UnnamedProject.dll
          obj/Debug/android/assets/Java.Interop.dll
          obj/Debug/android/assets/Mono.Android.dll
          obj/Debug/android/assets/mscorlib.dll
          obj/Debug/android/assets/System.Core.dll
          obj/Debug/android/assets/System.dll
          obj/Debug/android/assets/System.Xml.dll
          obj/Debug/android/assets/Xamarin.Android.Support.v4.dll
          obj/Debug/android/assets/Xamarin.Android.Support.v7.AppCompat.dll
          obj/Debug/android/assets/System.Runtime.dll
          obj/Debug/android/assets/System.ComponentModel.Composition.dll
          obj/Debug/android/assets/System.Diagnostics.Debug.dll
          obj/Debug/android/assets/System.Threading.dll
          obj/Debug/android/assets/System.Collections.dll
          obj/Debug/android/assets/System.Collections.Concurrent.dll
          obj/Debug/android/assets/System.Reflection.dll
          obj/Debug/android/assets/System.Linq.Expressions.dll
          obj/Debug/android/assets/System.Reflection.Primitives.dll
          obj/Debug/android/assets/System.Dynamic.Runtime.dll
          obj/Debug/android/assets/System.ObjectModel.dll
          obj/Debug/android/assets/System.Linq.dll
          obj/Debug/android/assets/System.Runtime.InteropServices.dll
          obj/Debug/android/assets/System.Runtime.Extensions.dll
          obj/Debug/android/assets/System.Reflection.Extensions.dll
          obj/Debug/android/assets/System.Net.Http.dll
          obj/Debug/android/assets/System.Runtime.Serialization.dll
          obj/Debug/android/assets/System.ServiceModel.Internals.dll
          obj/Debug/android/assets/Mono.Security.dll (TaskId:214)
  Task Parameter:
      ResolvedUserAssemblies=
          obj/Debug/android/assets/UnnamedProject.dll
          obj/Debug/android/assets/Xamarin.Android.Support.v4.dll
          obj/Debug/android/assets/Xamarin.Android.Support.v7.AppCompat.dll (TaskId:214)
  Task Parameter:PackagedResources=obj/Debug/android/bin/packaged_resources (TaskId:214)
  Task Parameter:
      DalvikClasses=
          obj/Debug/android/bin/dex/mono.android.dex
          obj/Debug/android/bin/classes.dex (TaskId:214)
  Task Parameter:PrimaryCpuAbi=x86 (TaskId:214)
  Task Parameter:External=False (TaskId:214)
  Task Parameter:ManifestTemplate=obj/Debug/android/AndroidManifest.xml (TaskId:214)
  Task Parameter:AndroidPlatformToolsDirectory=/Users/builder/android-toolchain/sdk/platform-tools/ (TaskId:214)
  Task Parameter:TargetFrameworkVersion=v9.0 (TaskId:214)
  Task Parameter:UseSharedRuntime=True (TaskId:214)
  Task Parameter:EnableInstantRun=True (TaskId:214)
  Task Parameter:InstantRunForceUploadApk=False (TaskId:214)
  Task Parameter:IntermediateOutputPath=obj/Debug/ (TaskId:214)
  Task Parameter:AndroidSdkPath=/Users/builder/android-toolchain/sdk (TaskId:214)
  Task Parameter:AndroidNdkPath=/Users/builder/android-toolchain/ndk (TaskId:214)
  Task Parameter:JavaSdkPath=/usr (TaskId:214)
  InstallPackageAssemblies Task (TaskId:214)
    AdbInstallOptions:  (TaskId:214)
    AdbOptions:  (TaskId:214)
    AdbTarget:  (TaskId:214)
    ApkFileSigned: bin/Debug/UnnamedProject.UnnamedProject-Signed.apk (TaskId:214)
    EmbedAssemblies: False (TaskId:214)
    PreserveUserData: True (TaskId:214)
    TargetFrameworkVersion: v9.0 (TaskId:214)
    UseSharedRuntime: True (TaskId:214)
    PackageName: UnnamedProject.UnnamedProject (TaskId:214)
    ResolvedAssemblies: (TaskId:214)
      obj/Debug/android/assets/UnnamedProject.dll (TaskId:214)
      obj/Debug/android/assets/Java.Interop.dll (TaskId:214)
      obj/Debug/android/assets/Mono.Android.dll (TaskId:214)
      obj/Debug/android/assets/mscorlib.dll (TaskId:214)
      obj/Debug/android/assets/System.Core.dll (TaskId:214)
      obj/Debug/android/assets/System.dll (TaskId:214)
      obj/Debug/android/assets/System.Xml.dll (TaskId:214)
      obj/Debug/android/assets/Xamarin.Android.Support.v4.dll (TaskId:214)
      obj/Debug/android/assets/Xamarin.Android.Support.v7.AppCompat.dll (TaskId:214)
      obj/Debug/android/assets/System.Runtime.dll (TaskId:214)
      obj/Debug/android/assets/System.ComponentModel.Composition.dll (TaskId:214)
      obj/Debug/android/assets/System.Diagnostics.Debug.dll (TaskId:214)
      obj/Debug/android/assets/System.Threading.dll (TaskId:214)
      obj/Debug/android/assets/System.Collections.dll (TaskId:214)
      obj/Debug/android/assets/System.Collections.Concurrent.dll (TaskId:214)
      obj/Debug/android/assets/System.Reflection.dll (TaskId:214)
      obj/Debug/android/assets/System.Linq.Expressions.dll (TaskId:214)
      obj/Debug/android/assets/System.Reflection.Primitives.dll (TaskId:214)
      obj/Debug/android/assets/System.Dynamic.Runtime.dll (TaskId:214)
      obj/Debug/android/assets/System.ObjectModel.dll (TaskId:214)
      obj/Debug/android/assets/System.Linq.dll (TaskId:214)
      obj/Debug/android/assets/System.Runtime.InteropServices.dll (TaskId:214)
      obj/Debug/android/assets/System.Runtime.Extensions.dll (TaskId:214)
      obj/Debug/android/assets/System.Reflection.Extensions.dll (TaskId:214)
      obj/Debug/android/assets/System.Net.Http.dll (TaskId:214)
      obj/Debug/android/assets/System.Runtime.Serialization.dll (TaskId:214)
      obj/Debug/android/assets/System.ServiceModel.Internals.dll (TaskId:214)
      obj/Debug/android/assets/Mono.Security.dll (TaskId:214)
    ResolvedUserAssemblies: (TaskId:214)
      obj/Debug/android/assets/UnnamedProject.dll (TaskId:214)
      obj/Debug/android/assets/Xamarin.Android.Support.v4.dll (TaskId:214)
      obj/Debug/android/assets/Xamarin.Android.Support.v7.AppCompat.dll (TaskId:214)
    NativeLibraries: (TaskId:214)
    DalvikClasses: (TaskId:214)
      obj/Debug/android/bin/dex/mono.android.dex (TaskId:214)
      obj/Debug/android/bin/classes.dex (TaskId:214)
    PackagedResources: (TaskId:214)
      obj/Debug/android/bin/packaged_resources (TaskId:214)
    SDCardDirectory:  (TaskId:214)
    External: False (TaskId:214)
    ManifestTemplate: obj/Debug/android/AndroidManifest.xml (TaskId:214)
    EnableInstantRun: True (TaskId:214)
    InstantRunForceUploadApk: False (TaskId:214)
    AndroidSdkPath: /Users/builder/android-toolchain/sdk (TaskId:214)
    AndroidNdkPath: /Users/builder/android-toolchain/ndk (TaskId:214)
    JavaSdkPath: /usr (TaskId:214)
``

Stuff is printed twice, the `Task Parameter` ones are MSBuild.

I'm all for just removing all these, as `LogMessage` is likely fast but not free.